cunumeric/module.py

```diff
@@ -6227,3 +6227,701 @@ def bincount(
+    midpoint:
+        NumPy method kept for backwards compatibility.
+        Uses ``(i + j) / 2``.
+    References
+    ----------
+    .. [1] R. J. Hyndman and Y. Fan,
+       "Sample quantiles in statistical packages,"
+       The American Statistician, 50(4), pp. 361-365, 1996
+    """
+
+    from .logic import isscalar
+
+    dict_methods = {
+        "inverted_cdf": inverted_cdf,
+        "averaged_inverted_cdf": averaged_inverted_cdf,
+        "closest_observation": closest_observation,
+        "interpolated_inverted_cdf": interpolated_inverted_cdf,
+        "hazen": hazen,
+        "weibull": weibull,
+        "linear": linear,
+        "median_unbiased": median_unbiased,
+        "normal_unbiased": normal_unbiased,
+        "lower": lower,
+        "higher": higher,
+        "midpoint": midpoint,
+        "nearest": nearest,
+    }
+
+    axes_set = []
+    original_shape = a.shape
+
+    if (axis is not None) and (not isscalar(axis)):
+        if len(axis) == 1:
+            real_axis = axis[0]
+            a_rr = a
+            axes_set = [real_axis]
+        else:
+            (real_axis, a_rr) = reshuffle_reshape(a, axis)
+            # What happens with multiple axes and overwrite_input = True ?
+            # It seems overwrite_input is reset to False;
+            overwrite_input = False
+            axes_set = axis
+    else:
+        real_axis = axis
+        a_rr = a
+        axes_set = [real_axis]
+
+    # covers both array-like and scalar cases:
+    #
+    q_arr = np.asarray(q)
+
+    # in the future k-sort (partition)
+    # might be faster, for now it uses sort
+    # arr = partition(arr, k = floor(nq), axis = real_axis)
+    # but that would require a k-sort call for each `q`!
+    # too expensive for many `q` values...
+    # if no axis given then elements are sorted as a 1D array
+    #
+    if overwrite_input:
+        a_rr.sort(axis=real_axis)
+        arr = a_rr
+    else:
+        arr = sort(a_rr, axis=real_axis)
+
+    # return type dependency on arr.dtype:
+    #
+    # it depends on interpolation method;
+    # For discontinuous methods returning either end of the interval within
+    # which the quantile falls, or the other; arr.dtype is returned;
+    # else, logic below:
+    #
+    # if is_float(arr_dtype) && (arr.dtype >= dtype('float64')) then
+    #    arr.dtype
+    # else
+    #    dtype('float64')
+    #
+    # see https://github.com/numpy/numpy/issues/22323
+    #
+    if method in [
+        "inverted_cdf",
+        "closest_observation",
+        "lower",
+        "higher",
+        "nearest",
+    ]:
+        to_dtype = arr.dtype
+    else:
+        to_dtype = (
+            arr.dtype
+            if (arr.dtype == np.dtype("float128"))
+            else np.dtype("float64")
+        )
+
+    res = quantile_impl(
+        arr,
+        q_arr,
+        real_axis,
+        axes_set,
+        original_shape,
+        dict_methods[method],
+        keepdims,
+        to_dtype,
+        out,
+    )
+
```

Comment:
`quantile(keepdims=True)` is not returning the correct shape:

```
print(np.quantile(np.random.random((3,4,5)), 0.5, keepdims=True).shape)
```

prints `(1, 1, 1)`, whereas

```
print(cn.quantile(cn.random.random((3,4,5)), 0.5, keepdims=True).shape)
```

prints `()`.

Reply:
The problem was `axis=None`. Fixed it in commits d09f4a5071a2077e601da7e9cdfaa04a150d747b and b2445ab2a3b393be0da873b550dffc017b2c03db. Added test.